src/metabase/search/scoring.clj

```diff
@@ -16,12 +16,21 @@
   [query :- s/Str]
   (str/lower-case query))
 
-(s/defn tokenize :- [s/Str]
-  "Break a search `query` into its constituent tokens"
+(s/defn tokenize
+  "Break a search `query` into its constituent tokens.
+   Don't add a return value schema to this function, it slows down the search."
   [query :- s/Str]
   (filter seq
           (str/split query #"\s+")))
 
+(s/defn sql-tokenize
+  "Break a search `query` into its constituent tokens for searching in a native SQL query
+   'foo.bar' is tokenized to ['foo.bar'] but 'foo(bar)' is tokenized to ['foo', 'bar'].
+   Don't add a return value schema to this function, it slows down the search."
+  [query :- s/Str]
+  (filter seq
+          (str/split query #"[\s+\(\)\[\]\{\}\,\'\"]")))
```

Comment:
this chokes on escaped weird tokens, doesn't it? like, you can have a query like:

`select * from "a))_tablename";`

and that'll select from the table that someone for some reason named `a))_tablename`

Reply:
You're right. I need a more general way to split strings that matches how to_tsvector('simple', <query>) does it. 

For example this is how tsvector tokenizes your query:
```
=# select to_tsvector('simple', 'select * from "a))_tablename"');
               to_tsvector               
-----------------------------------------
 'a':3 'from':2 'select':1 'tablename':4
(1 row)
```

I'll fix it and write the tests.